common/persistence/sql/dbal/class.Driver.php

```diff
@@ -216,10 +215,13 @@ public function quote($parameter, $parameter_type = PDO::PARAM_STR)
      */
     public function insert($tableName, array $data, array $types = [])
     {
-        $cleanColumns = [];
-        foreach ($data as $columnName => $value) {
-            $cleanColumns[$this->getPlatForm()->quoteIdentifier($columnName)] = $value;
+        $cleanColumns = $this->quoteColumnsMap($data);
+
+        if (is_string(key($types))) {
+            $types = $this->filterNotNumericParamTypes($types);
```

Comment:
if you always use the same name for the variable you are changing, you may get lost trying to understand the code.

Reply:
that's allows me to not duplicate values of $types variable in case if miss this `if clause`